test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,64 @@
+require "test_helper"
+
+describe UsersController do
+  before do
+    @user = User.create!(username: "test")
+  end
+
+  describe "index" do
+    it "can get the index page" do
+      get users_path
+
+      must_respond_with :success
+    end
+  end
+
+  describe "show" do
+    it "can get a valid users's page" do
+      get user_path(@user.id)
+
+      must_respond_with :success
+    end
+
+    it "will redirect for an invalid work id" do
+      get user_path(-1)
+
+      must_respond_with :redirect
+      must_redirect_to users_path
+    end
+  end
+
+  describe "login_form" do
+    it "can get the login page" do
+      get login_path
+
+      must_respond_with :success
+    end
+  end
+
+  describe "login" do
+    it "can login a user" do
+      user = perform_login
+
```

Comment:
There are a couple of behavior modes you're not covering:
- Logging in a new user (should create that user)
- Attempting to log in with an invalid username (e.g. empty string) (should fail)